File: arenagymapk/lib/models/usuario.dart
class Usuario {
  final int idUsuario;
  final String nombre;
  final String apellido;
  final String email;
  final String? fotoPerfil;
  final bool emailVerificado;
  final String? fechaNacimiento;
  final String? telefono;
  final String? fechaRegistro;
  final bool? activo;
  final String? ultimoAcceso;

  Usuario({
    required this.idUsuario,
    required this.nombre,
    required this.apellido,
    required this.email,
    this.fotoPerfil,
    this.emailVerificado = false,
    this.fechaNacimiento,
    this.telefono,
    this.fechaRegistro,
    this.activo,
    this.ultimoAcceso,
  });

  String get nombreCompleto => '$nombre $apellido';

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      idUsuario: json['id_usuario'] as int,
      nombre: json['nombre']?.toString() ?? '',
      apellido: json['apellido']?.toString() ?? '',
      email: json['email']?.toString() ?? '',
      fotoPerfil: json['foto_perfil'] as String?,
      emailVerificado: json['email_verificado'] as bool? ?? false,
      fechaNacimiento: json['fecha_nacimiento'] as String?,
      telefono: json['telefono'] as String?,
      fechaRegistro: json['fecha_registro'] as String?,
      activo: json['activo'] as bool?,
      ultimoAcceso: json['ultimo_acceso'] as String?,
    );
  }
}

File: frontend-gym/src/lib/db/models/usuario.js
const { DataTypes } = require('sequelize');
const sequelize = require('../sequelize');

const Usuario = sequelize.define('Usuario', {
  id_usuario: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  nombre: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  apellido: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  fecha_nacimiento: {
    type: DataTypes.DATEONLY,
    allowNull: true
  },
  telefono: {
    type: DataTypes.STRING(20),
    allowNull: true
  },
  email: {
    type: DataTypes.STRING(100),
    unique: true,
    allowNull: true
  },
  fecha_registro: {
    type: DataTypes.DATEONLY,
    allowNull: false,
    defaultValue: DataTypes.NOW
  },
  registrado_por: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'administrativos',
      key: 'id_admin'
    }
  },
  activo: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: true
  },
  // Campos para autenticación desde la futura app móvil de clientes.
  // No los usa (todavía) ningún endpoint del sistema web actual.
  password_hash: {
    type: DataTypes.STRING(255),
    allowNull: true
  },
  foto_perfil: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  ultimo_acceso: {
    type: DataTypes.DATE,
    allowNull: true
  },
  email_verificado: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: false
  }
}, {
  tableName: 'usuarios',
  timestamps: false,
  defaultScope: {
    where: {
      activo: true
    }
  },
  scopes: {
    withInactive: {
      where: {}
    },
    onlyInactive: {
      where: {
        activo: false
      }
    }
  }
});

module.exports = Usuario;
